libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -126,12 +138,21 @@ class CppInterface : public testing::Test {
     const auto compilation_result =
         compiler_.CompileGlslToSpv(shader, kind, "shader", options);
     EXPECT_TRUE(CompilationResultIsSuccess(compilation_result)) << kind << '\n';
-    // Use string(const char* s, size_t n) constructor instead of
-    // string(const char* s) to make sure the string has complete binary data.
-    // string(const char* s) assumes a null-terminated C-string, which will cut
-    // the binary data when it sees a '\0' byte.
-    return std::string(compilation_result.GetData(),
-                       compilation_result.GetLength());
+    // Need to make sure you get complete binary data, including embedded nulls.
+    return CompilerOutputAsString(compilation_result);
```

Comment:
There are tests later in the file for the presence of debug info.  Those tests use HasSubstr looking for the name of a variable from the GLSL source.  That's why we return the SPIR-V binary as a string.  We don't want to pattern-match against an OpName instruction since we don't want to overspecify the named ID.